crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -217,15 +217,37 @@ pub unsafe trait SystemParam: Sized {
     #[allow(unused_variables)]
     fn queue(state: &mut Self::State, system_meta: &SystemMeta, world: DeferredWorld) {}
 
-    /// Creates a parameter to be passed into a [`SystemParamFunction`].
+    /// Validates that the data can be acquired by [`get_param`](SystemParam::get_param).
+    /// The [`world`](UnsafeWorldCell) can only be used to read param's data
+    /// and world metadata. No data can be written.
+    /// This should be called before [`SystemParam::get_param`] when running systems
+    /// to ensure data can be acquired without panic.
```

Comment:
Similarly this should mention that `get_param` can get called even if `validate_param` returns `false`, and in that it should panic rather than resulting in UB.